src/run_task.py

```diff
@@ -62,52 +73,61 @@ def main(
     ] = False,
     datetime: Annotated[
-        str, typer.Option("--datetime", help="Datetime string (e.g. 2024)")
+        str,
+        typer.Option(
+            "--datetime",
+            help="Datetime string, either year or year period or specific date (e.g. 2024 or 2023-2024 or 2024-10-01)",
+        ),
     ] = "2024",
+    land_mask: Annotated[
+        str,
+        typer.Option(
+            "--land-mask",
+            help="Land mask to use osm, gadm or combined. Default is combined.",
+        ),
+    ] = "combined",
+    testrun: Annotated[
+        bool,
+        typer.Option(
+            "--testrun/--no-testrun",
+            help="Testrun: run without model inferencing to test data reading/writing",
+        ),
+    ] = False,
 ) -> None:
-
     log = logging.getLogger(tile_id)
-    log.info("Starting processing")
-    log.info(
-        f"Input received: \ntile_id: {tile_id}\n version: {version}\n output_bucket: {output_bucket}\n datetime: {datetime}\n overwrite: {overwrite}\n model_zip_uri: {model_zip_uri}\n collection_url_root: {collection_url_root}"
-    )
+
+    # ---- Setup common objects across dates to avoid reloading ----
 
     grid = PACIFIC_GRID_10
-    catalog = "https://stac.digitalearthpacific.org"
-    collection = "dep_s2_geomad"
+    catalog = "https://stac.dataspace.copernicus.eu/v1"
+    collection = "sentinel-2-global-mosaics"
+
+    assert (
+        collection_url_root == "https://stac.digitalearthpacific.org/collections"
+        and output_bucket == "dep-public-data"
+    ) or (
+        collection_url_root == "https://stac.staging.digitalearthpacific.io/collections"
+        and output_bucket == "dep-public-staging"
+    ), f"output_bucket={output_bucket} and collection_url_root={collection_url_root} not matched. Check the python run_task.py command"
+
+    if "staging" in collection_url_root:
+        profile = "staging"
+    else:
+        profile = "prod"
+
     # Make sure we can access S3
     log.info("Configuring S3 access")
-    configure_s3_access(cloud_defaults=True)
-    client = boto3.client("s3")
-
-    itempath = S3ItemPath(
-        bucket=output_bucket,
-        sensor="s2",
-        dataset_id="vegheight",
-        version=version,
-        time=datetime,
-    )
-
-    # tile_id is a string like "45,55"
-
-    stac_document = itempath.stac_path(tile_id)
-    log.info(
-        f"result will be saved to https://{output_bucket}.s3.us-west-2.amazonaws.com/{stac_document}"
-    )
-
-    # If we don't want to overwrite, and the destination file already exists, skip it
-    if not overwrite and object_exists(output_bucket, stac_document, client=client):
-        log.info(f"Item already exists at {stac_document}")
-        # This is an exit with success
-        raise typer.Exit()
-
-    ######################## where to upload model?
+    configure_s3_access(profile=profile, cloud_defaults=True)
+    session = boto3.Session(profile_name=profile)
+    client = session.client("s3")

```

Comment:
The configure_s3_access function creates a boto3 session.

https://odc-stac.readthedocs.io/en/latest/_api/odc.stac.configure_s3_access.html#odc.stac.configure_s3_access